lib/kvdb/sched_sts.c

```diff
@@ -62,21 +129,21 @@ sts_create(struct kvdb_rparams *rp, const char *name, uint nq, struct sts **hand
         return merr(ENOMEM);
 
     memset(self, 0, sizeof(*self));
-    self->sts_rp = rp;
-
-    snprintf(self->sts_name, sizeof(self->sts_name), "hse_sts_%u_%s",
-             rp->csched_policy, name);
-
-    if (csched_rp_dbg_mon(rp))
-        log_info("sts/mon create %s, queues %u, policy 0x%x",
-                 self->sts_name, nq, rp->csched_policy);
+    snprintf(self->sts_name, sizeof(self->sts_name), "hse_sts_%s", name);
```

Comment:
I have been checking the return value of snprintf. Here it seems like we can disregard truncation, but supposedly snprintf() can return < 0. What does that mean? I don't know! I normally bubble this up as EBADMSG

Reply:
Like you said, worst case it's truncated.  It's only used as the thread name, which can be seen by ps.  I really can't imagine it would ever fail, because if it could Linux probably couldn't finish booting :-)
So while I'm a huge fan of checking return code for things that matter whether they succeeded or not, adding error checks in cases like this just increase the overall cyclomatic complexity for dubios benefit..